include/FockSpace/FockSpace.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef GQCG_FOCKSPACE_HPP
+#define GQCG_FOCKSPACE_HPP
+
+
+#include "FockSpace/BaseFockSpace.hpp"
+#include <boost/numeric/conversion/converter.hpp>
+#include <boost/math/special_functions.hpp>
+
+
+namespace GQCG {
+
+
+class FockSpace: public GQCG::BaseFockSpace {
+private:
+    const size_t N;  // number of electrons
+    const size_t dim;  // dimension of the Fock space
+    Matrixu vertex_weights;
+
+
+    // PRIVATE METHODS
+    /**
+     *  In-place permute the representation, giving the next bitstring permutation in reverse lexical ordering.
+     *
+     *      Examples:
+     *          011 -> 101
+     *          101 -> 110
+     */
+    size_t ulongNextPermutation(size_t representation) ;
```

Comment:
this is not an in-place permutation, no pass-by-reference and no `void` return type. Remove excessive whitespace after declaration